packages/itwin/ec3-widget/src/test/TemplateMenu.test.tsx

```diff
@@ -0,0 +1,272 @@
+import { TemplateMenu } from "../components/TemplateMenu";
+import type { Configuration } from "../components/Template";
+
+const activeIModelConnection = moq.Mock.ofType<IModelConnection>();
+const reportsClient = moq.Mock.ofType<ReportsClient>();
+const configClient = moq.Mock.ofType<EC3ConfigurationsClient>();
+const oDataClient = moq.Mock.ofType<ODataClient>();
+
+jest.mock("@itwin/appui-react", () => ({
+  ...jest.requireActual("@itwin/appui-react"),
+  useActiveIModelConnection: () => activeIModelConnection.object,
+}));
+
+jest.mock("@itwin/itwinui-react", () => ({
+  ...jest.requireActual("@itwin/itwinui-react"),
+  toaster: {
+    positive: (_: string) => { },
+    negative: (_: string) => { },
+  },
+}));
+
+describe("TemplateMenu", () => {
+  const mockedReports = Array.from(
+    { length: 5 },
+    (_, index) => ({
+      id: index.toString(),
+      displayName: `report_${index}`,
+      description: "desc",
+      deleted: false,
+      _links: {
+        project: {
+          href: "reportLink",
+        },
+      },
+    })
+  );
+
+  const configId = "1234-1234-1234-1234";
+  const config: EC3Configuration = {
+    id: configId,
+    displayName: "mocked_configuration",
+    description: "mocked_description",
+    labels: [{
+      name: "mocked_label",
+      reportTable: "",
+      elementNameColumn: "",
+      elementQuantityColumn: "",
+      materials: [],
+    }],
+    createdBy: "",
+    modifiedBy: "",
+    createdOn: "",
+    modifiedOn: "",
+    _links: {
+      report: {
+        href: "link/reports/0",
+      },
+    },
+  };
+
+  const template: Configuration = {
+    id: configId,
+    displayName: "",
+    description: "",
+    labels: [],
+    reportId: "",
+  };
+
+  const iTwinId = faker.datatype.uuid();
+  const accessToken = faker.datatype.uuid();
+  const getAccessTokenFn = async () => accessToken;
+
+  beforeAll(async () => {
+    Element.prototype.scrollIntoView = jest.fn();
+    activeIModelConnection.setup((x) => x.iTwinId).returns(() => iTwinId);
+    reportsClient.setup(async (x) => x.getReports(accessToken, iTwinId)).returns(async () => mockedReports);
+    configClient.setup(async (x) => x.getConfiguration(accessToken, configId)).returns(async () => config);
+    configClient.setup(async (x) => x.updateConfiguration(accessToken, configId, moq.It.isAny())).returns(async () => config);
+    configClient.setup(async (x) => x.createConfiguration(accessToken, moq.It.isAny())).returns(async () => config);
+    oDataClient.setup(async (x) => x.getODataReportMetadata(accessToken, moq.It.isAny())).returns(async () => []);
+  });
+
+  it("Template Menu should render successfully for creating template", async () => {
+    await renderWithContext({
+      component: <TemplateMenu
+        goBack={async () => { }}
+        created={false}
+      />,
+    });
+    expect(screen.getByTestId("ec3-templateDetails")).toBeDefined();
```

Comment:
Inconsistent `test-id` - one uses dashes to separate words, another uses camel casing.